### PR TITLE
Fixed cddl spec for tx-submission protocol

### PR DESCRIPTION
**Pull request by coot**
We should use `tx` rather than `txId` in the cddl.  It only works
because we use `int` for both.

### CODE REVIEW

**Review by bolt12 — APPROVED** [review]

**Comment by coot** [comment]
@bolt12 I changed this PR considerably, so please review it again.

We should use this idea in other mini-protocols / specs as well.  The only question is if we should modify data structure in `ouroboros-network-protocols:test` or provide different data structures just for `ouroboros-network-protocols:cddl`?

**Review by bolt12 — CHANGES_REQUESTED** [review]
Nice, is there an open issue about this? Would be nice to track this nice improvement

File: ouroboros-network-protocols/testlib/Ouroboros/Network/Protocol/TxSubmission2/Test.hs [hunk 1]
```diff
@@ -94,8 +96,22 @@ instance ShowProxy Tx where
...
+    arbitrary = oneof [ TxId . CBOR.TInt <$> arbitrary
+                      , TxId . CBOR.TBytes <$> arbitrary
+                      , TxId . CBOR.TBool <$> arbitrary
+                      ]
+    shrink (TxId term) =
+      case term of
+        CBOR.TInt a   -> TxId . CBOR.TInt   <$> shrink a
+        CBOR.TBytes a -> TxId . CBOR.TBytes <$> shrink a
+        CBOR.TBool a  -> TxId . CBOR.TBool  <$> shrink a
+        _             -> []
```
[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network-protocols/testlib/Ouroboros/Network/Protocol/TxSubmission2/Test.hs:114]: I was thinking rather to have `Any` newtype wrapper, but that's only useful if we do this kind of thing for other mini-protocols too.

**Comment by coot** [comment]
I created the issue: input-output-hk/ouroboros-network#4595

File: ouroboros-network-protocols/testlib/Ouroboros/Network/Protocol/TxSubmission2/Test.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bolt12) [reply ouroboros-network-protocols/testlib/Ouroboros/Network/Protocol/TxSubmission2/Test.hs:114]: Yes that makes sense :)

File: ouroboros-network-protocols/test-cddl/specs/common.cddl [hunk 2]
```diff
@@ -21,8 +21,11 @@ origin          = []
 blockHeaderHash = [slotNo, int]
 slotNo          = word64
 
-txId         = int
-transaction  = int
+; In this spec we don't specify what transaction idenfiers (txId) or
+; transactions (tx) are themselves, `ouroboros-network` is polymorphic over
+; them.
+txId         = any
+tx           = any
```
Comment (oskin1) [inline ouroboros-network-protocols/test-cddl/specs/common.cddl:28]: Where can I find specification of what is expected here by the actual cardano-node implementation?

Comment (oskin1) [reply ouroboros-network-protocols/test-cddl/specs/common.cddl:28]: @coot

**Pull request merged**